pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -760,7 +760,7 @@ private void checkReplicatedSubscriptionMarker(Position position, MessageImpl<?>
 
         int markerType = msg.getMessageBuilder().getMarkerType();
 
-        if (!remoteCluster.equals(msg.getMessageBuilder().getReplicatedFrom())) {
+        if (!remoteCluster.equals(msg.getReplicatedFrom())) {
```

Comment:
I am not sure this is the right fix. 

If you want to address the issue `java.lang.IllegalStateException: Field 'replicated_from' is not set`, you should check `msg.getMessageBuilder().hasReplicatedFrom()`.

Your change brings inconsistent usage across different places. Line 756 and 761 are using `MessageBuilder` and Line 763 is using `Message`.

Reply:
Changed as suggested.